packages/runtime-common/current-run.ts

```diff
@@ -159,16 +178,22 @@ export class CurrentRun {
     instances?: URLMap<SearchEntryWithErrors>;
     modules?: Map<string, ModuleWithErrors>;
     ignoreMap?: URLMap<Ignore>;
-    loader?: LoaderType;
+    loader?: Loader;
   }) {
     this.#realmPaths = new RealmPaths(realm.url);
     this.#reader = reader;
     this.realm = realm;
     this.#instances = instances;
     this.#modules = modules;
     this.#ignoreMap = ignoreMap;
-    this.#loader =
-      loader ?? (Loader.createLoaderFromGlobal() as unknown as LoaderType);
+    this.#loader = loader ?? Loader.createLoaderFromGlobal();
+    this.#fastboot = realm.makeFastBoot
+      ? realm.makeFastBoot(
+          this.loader.getUrlHandlers(),
+          this.loader.getUrlMappings(),
+          this.#staticResponses
+        )
```

Comment:
instead of making a new fastboot for each current run, we could use the ember app's loader service to reset the  current loader in the existing fastboot instance (which is a thing we already support for editing card source files). this could dramatically speed up the indexing and be a much more efficient use of fastboot.